dvc/remote/gdrive/client.py

```diff
@@ -0,0 +1,284 @@
+from time import sleep
+import logging
+import posixpath
+import os
+
+from funcy import cached_property
+
+from requests import ConnectionError
+
+from dvc.progress import progress
+from dvc.remote.gdrive.utils import (
+    metadata_isdir,
+    response_is_ratelimit,
+    MIME_GOOGLE_APPS_FOLDER,
+    response_error_message,
+)
+from dvc.remote.gdrive.exceptions import (
+    GDriveError,
+    GDriveHTTPError,
+    GDriveResourceNotFound,
+)
+from dvc.remote.gdrive.oauth2 import OAuth2
+
+logger = logging.getLogger(__name__)
+
+
+class GDriveClient:
+
+    GOOGLEAPIS_BASE_URL = "https://www.googleapis.com/"
+    TIMEOUT = (5, 60)
+
+    def __init__(
+        self,
+        space,
+        oauth_id,
+        credentialpath,
+        scopes,
+        oauth2_flow_runner,
+        max_retries=10,
+    ):
+        self.space = space
+        self.oauth_id = oauth_id
+        self.credentialpath = credentialpath
+        self.scopes = scopes
+        self.oauth2_flow_runner = oauth2_flow_runner
+        self.max_retries = max_retries
+        self.oauth2 = OAuth2(
+            oauth_id, credentialpath, scopes, oauth2_flow_runner
+        )
+
+    @cached_property
+    def session(self):
+        """AuthorizedSession to communicate with https://googleapis.com
+
+        Security notice:
```

Comment:
it's probably easy to add a test/assert to check that domain is intact

Reply:
Great idea, thanks! What do you think if I'd just override the `request()` of `AuthorizedSession`?